brave/src/main/java/brave/baggage/BaggagePropagation.java

```diff
@@ -331,54 +320,55 @@ public FactoryBuilder add(BaggagePropagationConfig config) {
       BaggageFields extra = context.findExtra(BaggageFields.class);
       if (extra == null) return;
       Map<String, String> values =
-          extra.toMapFilteringFieldNames(propagation.factory.localFieldNames);
+          extra.toMapFilteringFieldNames(factory.localFieldNames);
       if (values.isEmpty()) return;
 
-      for (BaggageCodecWithKeys<K> baggageCodecWithKeys : propagation.baggageCodecWithKeys) {
-        String value = baggageCodecWithKeys.baggageCodec.encode(values, context, request);
+      for (BaggagePropagationConfig config : factory.configs) {
+        if (config.baggageCodec == BaggageCodec.NOOP) continue; // local field
```

Comment:
Probably just dejavu but I feel as if I've seen this `if` statement on the right hand side of 3 PRs now :O

Reply:
haha tell me about it.. cleanup of this will come.. I just wanted to get rid of the K first. thanks for your patience with the road work!